### PR TITLE
Transfer dialog improvements & delegation.

### PR DESCRIPTION
**Pull request by talhasch**
@feruzm Please do not merge this PR now. It is better to merge once all missing wallet features done.

Defined different success messages per mode. Please review/check them in locale file.

### CODE REVIEW

File: src/common/components/delegated-vesting/index.tsx [hunk 1]
```diff
@@ -24,149 +33,147 @@ import parseAsset from "../../helper/parse-asset";
...
+                                },
+                                onKey: (key) => {
+                                    this.stateSet({inProgress: true});
+                                    delegateVestingShares(activeUser.username, key, username, "0.000000 VESTS")
+                                        .then(() => this.fetch())
+                                        .catch(err => error(formatError(err)))
+                                        .finally(() => this.stateSet({inProgress: false}))
+                                },
+                                onHot: () => {
+                                    delegateVestingSharesHot(activeUser.username, username, "0.000000");
```
Comment (feruzm) [inline src/common/components/delegated-vesting/index.tsx:132]: I believe `vesting_shares` should be in `0.000000 VESTS`, just numbers wouldn't work.

File: src/common/components/transfer/index.tsx [hunk 2]
```diff
@@ -377,26 +420,29 @@ export class Transfer extends Component<Props, State> {
...
+            case "withdraw-saving":
+                transferFromSavingsHot(username, to, fullAmount, memo);
                 break;
-            case 'power-up':
-                prms = transferToVestingHot(username, to, fullAmount);
+            case "power-up":
+                transferToVestingHot(username, to, fullAmount);
+                break;
+            case "delegate":
+                const vests = this.hpToVests(Number(amount), false);
```
Comment (feruzm) [inline src/common/components/transfer/index.tsx:444]: `withSymbol` should be true otherwise we would get serialization issue.

**Review by feruzm — CHANGES_REQUESTED** [review]

File: src/common/components/transfer/index.tsx (hunk 2 above)
Comment (talhasch) [reply src/common/components/transfer/index.tsx:444]: hive signer doesn't accept with amount with `VESTS` that's why withSymbol exists in that function.

Comment (feruzm) [reply src/common/components/transfer/index.tsx:444]: @talhasch Really, I tried without VESTS with hivesigner, it said serialization issue.

Comment (feruzm) [reply src/common/components/transfer/index.tsx:444]: here screenshot of error: 
image

Comment (talhasch) [reply src/common/components/transfer/index.tsx:444]: ok let me check it

**Comment by talhasch** [comment]
@feruzm can you check power down?

**Comment by feruzm** [comment]
@talhasch ok

**Comment by feruzm** [comment]
@talhasch so far my comments

- Delegation shouldn't allow to own profile
image

- Power down looks good and simple, but we need setting route as well like we used to have.

**Comment by talhasch** [comment]
@feruzm  added withdraw routes into the dropdown on hive power section on wallet page.

Delegation used to set active user as target user by default. Fixed that. If user wants to delegate themself anyway, blockchain rejects.

Doing some more checks. Then ready to merge this.

**Comment by talhasch** [comment]
@feruzm ready to merge

**Review by feruzm — APPROVED** [review]
lgtm, good call on cleaning up pre-hardfork variables.

**Pull request merged**